choreographer/src/main/java/eu/arrowhead/core/choreographer/database/service/ChoreographerExecutorDBService.java

```diff
@@ -0,0 +1,279 @@
+import eu.arrowhead.common.database.repository.ChoreographerExecutorRepository;
+import eu.arrowhead.common.database.repository.ChoreographerExecutorServiceDefinitionRepository;
+import eu.arrowhead.common.database.repository.ChoreographerSessionStepRepository;
+import eu.arrowhead.common.dto.internal.ChoreographerExecutorListResponseDTO;
+import eu.arrowhead.common.dto.internal.ChoreographerStatusType;
+import eu.arrowhead.common.dto.internal.DTOConverter;
+import eu.arrowhead.common.dto.shared.ChoreographerExecutorResponseDTO;
+import eu.arrowhead.common.exception.ArrowheadException;
+import eu.arrowhead.common.exception.InvalidParameterException;
+import eu.arrowhead.common.verifier.CommonNamePartVerifier;
+
+@Service
+public class ChoreographerExecutorDBService {
+
+	//=================================================================================================
+    // members
+	
+	@Autowired
+    private ChoreographerExecutorRepository executorRepository;
+	
+	@Autowired
+	private ChoreographerExecutorServiceDefinitionRepository executorServiceDefinitionRepository;
+	
+	@Autowired
+	private ChoreographerSessionStepRepository sessionStepRepository;
+	
+	@Autowired
+	private CommonNamePartVerifier cnVerifier;
+	
+	private final Logger logger = LogManager.getLogger(ChoreographerExecutorDBService.class);
+	
+	//=================================================================================================
+    // methods
+
+	//-------------------------------------------------------------------------------------------------
+	public ChoreographerExecutorResponseDTO createExecutorResponse(final String systemName, final String address, final int port, final String baseUri, final String serviceDefinitionName,
+																   final int minVersion, final int maxVersion) { //TODO junit
+		logger.debug("createExecutorResponse started...");
+		
+		final ChoreographerExecutor executor = createExecutor(systemName, address, port, baseUri, serviceDefinitionName, minVersion, maxVersion);		
+		final List<ChoreographerExecutorServiceDefinition> serviceDefinitions = executorServiceDefinitionRepository.findAllByExecutor(executor);
+		
+		return DTOConverter.convertExecutorToExecutorResponseDTO(executor, serviceDefinitions);
+	}
+	
+    //-------------------------------------------------------------------------------------------------
+	@Transactional(rollbackFor = ArrowheadException.class)
+    public ChoreographerExecutor createExecutor(final String systemName, final String address, final int port, final String baseUri, final String serviceDefinitionName,
+    											final int minVersion, final int maxVersion) { //TODO junit
+		logger.debug("createExecutor started...");
+		Assert.isTrue(!Utilities.isEmpty(systemName), "systemName is empty");
+		Assert.isTrue(!Utilities.isEmpty(address), "address is empty");
+		// System name and address is coming from SR registration result, so should be verified
```

Comment:
it is a public method. Maybe the name and address comes from SR now, but who knows the future?

Reply:
My concern is that for example the address validation behavior is depending on the app property settings of SR and didn't want to call web service from this DB layer.
What if I rename the method to `createExecutorWithoutSystemNameAndAddressVerification` ?

Reply:
and add comment about the reason